crates/gitbutler-diff/src/write.rs

```diff
@@ -178,6 +185,22 @@ where
                 let new_blob_oid = git_repository.blob(blob_contents.as_bytes())?;
                 // upsert into the builder
                 builder.upsert(rel_path, new_blob_oid, filemode);
+            } else if !full_path.exists()
+                && hunks.len() == 1
+                && hunks[0].change_type == crate::ChangeType::Added
+            {
+                // File was deleted but now that hunk is being discarded with an inversed hunk
```

Comment:
Ah, I see now that the single-hunk mode is happening when the UI discards things.